src/src/rx-serial/SerialHoTT_TLM.cpp

```diff
@@ -0,0 +1,379 @@
+#if defined(TARGET_RX) && (defined(PLATFORM_ESP8266) || defined(PLATFORM_ESP32))
+
+#include "telemetry.h"
+#include "FIFO.h"
+#include "SerialHoTT_TLM.h"
+
+#define HOTT_POLL_RATE      150     // default HoTT bus poll rate [ms]
+#define HOTT_LEAD_OUT       10      // minimum gap between end of payload to next poll
+
+#define DISCOVERY_TIMEOUT   30000   // 30s device discovery time
+
+#define VARIO_MIN_CRSFRATE  1000    // CRSF telemetry packets will be sent if
+#define GPS_MIN_CRSFRATE    5000    // min rate timers in [ms] have expired
+#define BATT_MIN_CRSFRATE   5000    // or packet value has changed. Fastest to
+                                    // be expected update rate will by about 150ms due
+                                    // to HoTT bus speed if only a HoTT Vario is connected and
+                                    // values change every HoTT bus poll cycle.
+
+typedef struct crsf_sensor_gps_s {
+    int32_t latitude;               // degree / 10,000,000 big endian
+    int32_t longitude;              // degree / 10,000,000 big endian
+    uint16_t groundspeed;           // km/h / 10 big endian
+    uint16_t heading;               // GPS heading, degree/100 big endian
+    uint16_t altitude;              // meters, +1000m big endian
+    uint8_t satellites;             // satellites
+} PACKED crsf_sensor_gps_t;
+
+extern Telemetry telemetry;
+
+
+int SerialHoTT_TLM::getMaxSerialReadSize()
+{
+    return HOTT_MAX_BUF_LEN - hottInputBuffer.size();
+}
+
+void SerialHoTT_TLM::processBytes(uint8_t *bytes, u_int16_t size)
+{
+    hottInputBuffer.pushBytes(bytes, size);
+}
+
+void SerialHoTT_TLM::sendQueuedData(uint32_t maxBytesToSend)
+{  
+    uint32_t now = millis();
+
+    // device discovery timer
+    if(discoveryMode && (now - discoveryTimerStart >= DISCOVERY_TIMEOUT))
+        discoveryMode = false;
+
+    // device polling scheduler
+    if(now - lastPoll >= HOTT_POLL_RATE) {          
+        lastPoll = now; 
+
+        // start up in device discovery mode, after timeout regular operation 
+        pollNextDevice();
+    }
+
+    uint8_t size = hottInputBuffer.size(); 
+
+    if(size == sizeof(hottBusFrame)) {
+        // frame complete, prepare to poll next device after lead out time elapsed
+        lastPoll = now - HOTT_POLL_RATE + HOTT_LEAD_OUT;
+
+        //fetch received serial data
+        hottInputBuffer.popBytes((uint8_t *)&hottBusFrame, size);
+
+        // process received frame if CRC is ok
+        if(hottBusFrame.payload[CRC_INDEX] == calcFrameCRC((uint8_t *)&hottBusFrame.payload))
+            processFrame();
```

Comment:
curlys, I'll stop mentioning these now 😄

Reply:
done and done